scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,261 @@
+package plugin
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	schedulingresultstore "github.com/kubernetes-sigs/kube-scheduler-simulator/scheduler/plugin/resultstore"
+)
+
+//go:generate mockgen -destination=./mock/$GOFILE -package=plugin . Store,FilterPluginExtender,ScorePluginExtender,NormalizeScorePluginExtender
+//go:generate mockgen -destination=./mock/framework.go -package=plugin k8s.io/kubernetes/pkg/scheduler/framework FilterPlugin,ScorePlugin,ScoreExtensions
+type Store interface {
+	AddNormalizedScoreResult(namespace, podName, nodeName, pluginName string, normalizedscore int64)
+	AddFilterResult(namespace, podName, nodeName, pluginName, reason string)
+	AddScoreResult(namespace, podName, nodeName, pluginName string, score int64)
+}
+
+type FilterPluginExtender interface {
```

Comment:
ditto for other extenders.

```suggestion
// FilterPluginExtender is the extender for Filter plugin.
type FilterPluginExtender interface {
```